src/Ng/Jobs/NgJob.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Ng.Jobs
+{
+    public abstract class NgJob
+    {
+        protected ILoggerFactory LoggerFactory;
+        protected ILogger Logger;
+
+        protected NgJob(ILoggerFactory loggerFactory)
+        {
+            LoggerFactory = loggerFactory;
+
+            // We want to make a logger using the subclass of this job.
+            // Use reflection to find the type that is being constructed and create a logger with its type.
+            var derivedType = GetType();
+            derivedType.GetMethod(nameof(InitializeLogger))
+                .MakeGenericMethod(derivedType)
```

Comment:
discovered that `ILoggerFactory` has a `CreateLogger(Type t)` so i will use it instead of `CreateLogger<T>()` which gets rid of the reflection!